api/tests/test_views.py

```diff
@@ -89,3 +100,86 @@ def test_project_detail_delete(self):
         )
         self.assertEqual(api_response.status_code, HTTP_204_NO_CONTENT)
         self.assertEqual(Project.objects.count(), 0)
+
+
+class StationApiTests(APITestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        self.existing_project = ProjectFactory.create()
+        self.station = StationFactory.build(project=self.existing_project)
+        self.station_data = StationSerializer(self.station).data
+        self.station_list_url = reverse('station-list')
+
+    def create_station(self):
+        self.client.login(username=self.user.username, password=UserFactory.DEFAULT_PASSWORD)
+        return self.client.post(self.station_list_url, self.station_data, format='json')
+
+    def assertStationDataEqual(self, data):
+        self.assertEqual(data['name'], self.station_data['name'])
+        self.assertEqual(data['type'], self.station_data['type'])
+        self.assertEqual(data['notes'], self.station_data['notes'])
+        self.assertEqual(data['is_in_test_mode'], self.station_data['is_in_test_mode'])
+        self.assertEqual(data['altitude'], self.station_data['altitude'])
+        self.assertEqual(data['position'], self.station_data['position'])
+        self.assertEqual(data['country'], self.station_data['country'])
+        self.assertEqual(data['state'], self.station_data['state'])
+        self.assertEqual(data['county'], self.station_data['county'])
+        self.assertEqual(data['community'], self.station_data['community'])
+        self.assertEqual(data['city'], self.station_data['city'])
+        self.assertEqual(data['district'], self.station_data['district'])
+        self.assertEqual(data['project'], self.station_data['project'])
+        self.assertEqual(data['last_metering'], self.station_data['last_metering'])
+
+    def test_station_create(self):
+        self.assertEqual(Station.objects.count(), 0)
+        api_response = self.create_station()
+        self.assertEqual(api_response.status_code, HTTP_201_CREATED)
+        self.assertEqual(Station.objects.count(), 1)
+
+        created_station = Station.objects.get()
+        created_station_data = StationSerializer(created_station).data
+        self.assertStationDataEqual(created_station_data)
+        self.assertEqual(created_station.owner, self.user)
+
+    def test_add_metering(self):
+        station = StationFactory.create()
+        station_api_url = reverse('station-detail', kwargs={'pk': station.pk})
+
+        metering = MeteringFactory.build(station=None)
+        metering_data = MeteringSerializer(metering).data
+
+        # test cache key removal before add_metering
+        self.assertEqual(station.last_metering, MeteringSerializer(None).data)
+
+        self.assertEqual(station.metering_set.count(), 0)
+        add_metering_api_url = '{}add_metering/?token={}'.format(station_api_url, station.token)
```

Comment:
:+1: it's hidden inside drf documentation how to do it properly, but manage to find it: http://www.django-rest-framework.org/api-guide/routers/#extra-link-and-actions